### ISSUE TITLE
Wiki: Add random page as quick reply

### ISSUE THREAD

**Issue by swapagarwal** [swapagarwal/JARVIS-on-Messenger#248]

**Comment by marshareb** [comment]
What do you mean as a quick reply?

**Comment by edadesd** [comment]
Quick Replies add buttons to a response that a user can click for effect:

https://developers.facebook.com/docs/messenger-platform/send-api-reference/quick-replies

**Comment by aviral1701** [comment]
Hi @swapagarwal !
I would like on this one! Can I take it? 
Also instead of any random wiki page, we could add pages given in the "See Also" section of the wiki pages. Is this better or should I stick with any random page?

**Comment by aviral1701** [comment]
@swapagarwal Any Updates?

**Comment by made2591** [comment]
@aviral1701 me too, can I work at this with you? Have you already forked?

**Comment by rahul0379** [comment]
Hi @swapagarwal!

I would also like to work on this.
I was looking throught the code of `wiki.py` and it seems that the current deployment of JARVIS does not catch the `wikipedia.exceptions.DisambiguationError`. For example, input "wikipedia shiv"(wiki shiv leads to a disambiguation page) to Jarvis should return buttons for all the options. Instead, it goes to the generic exception and says 'I couldn't find any wikipedia results matching your query.' Just a version difference between current code and deployed code?

**Comment by swapagarwal** [comment]
As @marshareb was the first to get involved, the first option goes to him. 🎉
@marshareb Would you be willing to own this?
@edadesd Thanks!
@aviral1701 Did you look into the API to see if we could get the "See Also" pages? That would definitely be better. 😄 
@rahul0379 The code is in sync. 🤔 Can you try again? I'm getting the expected results.
@aviral1701 @made2591 @rahul0379 I have some issues noted down. I'll add them to the repo soon. Feel free to pick any of those! 😅

**Comment by made2591** [comment]
@swapagarwal no problems at all XD I'm currently working on several small-medium-big projects so...let's code 👍

**Comment by aviral1701** [comment]
@swapagarwal Yes! Here is a good explanation : https://stackoverflow.com/questions/37396367/how-to-extract-links-from-see-also-section-of-an-article-in-wikipedia
Let me know if I can help in anyway!

**Comment by swapagarwal** [comment]
We use the MediaWiki API. Will this require a migration?

**Comment by aviral1701** [comment]
@swapagarwal I think we should keep migration as the last resort.Few conclusions:

1. The MediaWiki Api doesn't offer the See Also section of the wiki page.
2. We can use Section attribute of the page, but it doesn't work with all pages.
3. We take the **html of the page and parse it**.Then find links in "See Also" div of the page.Might make the program considerably slow.
4. We use just the wikipedia API to get the see also part.
5. Or we can stick with giving back the random pages.
Please suggest what should be done here.

**Comment by swapagarwal** [comment]
Let's avoid parsing. We can add both random and see also pages.

**Comment by aviral1701** [comment]
#292

**Comment by dylmye** [comment]
Am I good to finish off #292?

**Comment by dbalakirev** [comment]
Hey,
If it's fine, I'd like to work on this. I prepared myself some tavern tests to ensure quality (maybe that is something I could contribute in a different issue?).
Cheers

**Comment by dbalakirev** [comment]
So here we go:
https://github.com/swapagarwal/JARVIS-on-Messenger/pull/946
